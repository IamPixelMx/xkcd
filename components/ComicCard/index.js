import ComicCard from './ComicCard';

export { ComicCard };
