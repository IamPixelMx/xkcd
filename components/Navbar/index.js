import Navbar from './Navbar';
import NavbarInput from './NavbarInput';
import NavbarItem from './NavbarItem';

export { Navbar, NavbarInput, NavbarItem };
