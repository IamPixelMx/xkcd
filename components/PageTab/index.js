import PageTab from './PageTab';

export { PageTab };
