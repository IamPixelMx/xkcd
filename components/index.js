import { Footer } from './Footer';
import { Navbar } from './Navbar';
import { PageTab } from './PageTab';

export { Footer, Navbar, PageTab };
