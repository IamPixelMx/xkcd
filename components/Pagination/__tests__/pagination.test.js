import React from 'react';
import * as ReactReduxHooks from 'react-redux';
import configureStore from 'redux-mock-store';
import { fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import { renderWithProvider, renderWithProviderSnapshot } from 'jest-utils';
import { Pagination } from 'components';

const dispatch = jest.fn();
jest.spyOn(ReactReduxHooks, 'useDispatch').mockImplementation(() => dispatch);

const mockStore = configureStore([]);

describe('Pagination', () => {
  it('Renders Pagination ', () => {
    const paginationElement = renderWithProviderSnapshot(<Pagination />, {
      mockStore,
    }).toJSON();
    expect(paginationElement).toMatchSnapshot();
  });

  it('Call dispatch on "prev" and "next" buttons', async () => {
    const { getByTitle } = renderWithProvider(<Pagination />, {
      mockStore,
    });
    const nextButton = getByTitle('next');
    const prevButton = getByTitle('prev');
    await act(async () => {
      fireEvent.click(nextButton);
      fireEvent.click(prevButton);
    });
    expect(dispatch).toBeCalledTimes(2);
  });
});
