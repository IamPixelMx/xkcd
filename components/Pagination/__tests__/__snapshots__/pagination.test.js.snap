// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders Pagination  1`] = `
<nav
  aria-label="pagination"
  className="pagination column is-rounded"
  role="navigation"
>
  <button
    className="pagination-previous button is-dark"
    onClick={[Function]}
    title="prev"
  >
    Siguiente
  </button>
  <button
    className="pagination-next button is-dark"
    onClick={[Function]}
    title="next"
  >
    Anterior
  </button>
</nav>
`;
